### DISCUSSION TITLE
Active Health check set https, failed SSL handshake self signed certificate in certificate chain

### DISCUSSION THREAD

**Discussion by yulogs** [Kong/kong#8689] [Help]
When setting Active Health check between Kong and the upstream server and using https( https_verify_certificate=true),
error message: failed SSL handshake self signed certificate in certificate chain.
I call admin api to trusted target server CA, but not ok..
(admin api : POST /ca_certificates)
Help me what can I set to trust targer server, thank you..

**Comment by dela1974** [comment]
Hi, I have not been able to reproduce this issue here. Can you give me more detailed information about your configuration

**Comment by yulogs** [comment]
error message:
2022/04/15 12:29:47 [error] 24#0: *54582 [lua] healthcheck.lua:1123: log(): [healthcheck] (483cf005-b90e-4375-a4f5-a2e614f47769:xxxxx-upstream) failed SSL handshake with 'xxx.xxx.xxx (xxx.xxx.xxx.xxx:443)': 19: self signed certificate in certificate chain, context: ngx.timer

my configuration

POST /ca_certificates
request:
{
    "cert":"-----BEGIN CERTIFICATE------................"
}

get /upstream
response:
{
    "hash_fallback_header": null,
    "algorithm": "round-robin",
    "tags": null,
    "name": "xxxxx-upstream",
    "hash_on": "none",
    "created_at": 1650443208,
    "healthchecks": {
        "passive": {
            "type": "http",
            "unhealthy": {
                "http_statuses": [
                    429,
                    500,
                    503
                ],
                "tcp_failures": 0,
                "timeouts": 0,
                "http_failures": 0
            },
            "healthy": {
                "successes": 1,
                "http_statuses": [
                    200,
                    201,
                    202,
                    203,
                    204,
                    205,
                    206,
                    207,
                    208,
                    226,
                    300,
                    301,
                    302,
                    303,
                    304,
                    305,
                    306,
                    307,
                    308
                ]
            }
        },
        "threshold": 0,
        "active": {
            "type": "https",
            "concurrency": 10,
            "healthy": {
                "interval": 30,
                "http_statuses": [
                    200,
                    302
                ],
                "successes": 1
            },
            "timeout": 1,
            "unhealthy": {
                "interval": 10,
                "tcp_failures": 0,
                "timeouts": 0,
                "http_failures": 10,
                "http_statuses": [
                    429,
                    404,
                    500,
                    501,
                    502,
                    503,
                    504,
                    505
                ]
            },
            "http_path": "/actuator/health",
            "https_sni": null,
            "https_verify_certificate": true
        }
    },
    "hash_on_header": null,
    "slots": 1000,
    "hash_fallback": "none",
    "host_header": null,
    "client_certificate": null,
    "hash_on_cookie": null,
    "hash_on_cookie_path": "/",
    "id": "a9649704-3ceb-4c66-b5d9-5eba4608a6b9"
}

get /upstreams/xxxxx-upstream/targets
response:
{
    "next": null,
    "data": [
        {
            "created_at": 1650443208.735,
            "weight": 50,
            "upstream": {
                "id": "a9649704-3ceb-4c66-b5d9-5eba4608a6b9"
            },
            "tags": null,
            "target": "xxx.xxx.xxx:443",
            "id": "59758081-5002-4b83-944c-484a93d13ac9"
        }
    ]
}
